app/controllers/pages_controller.rb

```diff
@@ -1,19 +1,29 @@
 class PagesController < ApplicationController
-    before_action :authenticate
+  before_action :authenticate
 
-    def index
+  def index
 
-    end
+  end
 
-    private
+  private
 
-    def authenticate
-        if session.has_key? :current_user_id and Student.find_by_id(session[:current_user_id])
-            @logged_out = false
-            student = Student.find_by_id(session[:current_user_id])
-            @name = student.first_name + " " + student.last_name
-        else
-            @logged_out = true
-        end
+  def authenticate
+    @user_type = Array.new
+    user = User.find_by_id(session[:current_user_id])
+    if user
+      @logged_out = false
+      @name = user.first_name + " " + user.last_name   
+      if user.is_admin
+        @user_type.push 'Admin'
```

Comment:
Not sure if a user can have multiple identities (e.g. a student can also be an admin), so make `@user_type` an array